test/unit/tracks/text-track-controls.test.js

```diff
@@ -262,3 +262,152 @@ if (!browser.IS_IE8) {
+  });
+
+  this.clock.tick(1000);
+
+  ok(player.controlBar.chaptersButton.hasClass('vjs-hidden'), 'chapters menu is not displayed');
+  equal(player.textTracks().length, 1, 'textTracks contains one item');
+
+  player.dispose();
+});
+
+test('chapters should be displayed when cues added to initial track and button updated', function() {
+  let player = TestHelpers.makePlayer({
+    tracks: [chaptersTrack]
+  });
+
+  this.clock.tick(1000);
+  
+  var chapters = player.textTracks()[0];
+  chapters.addCue({
+    startTime: 0,
+    endTime: 2,
+    text: 'Chapter 1'
+  });
+  chapters.addCue({
+    startTime: 2,
+    endTime: 4,
+    text: 'Chapter 2'
+  });
+  equal(chapters.cues.length, 2);
+
+  player.controlBar.chaptersButton.update();
+  
+  ok(!player.controlBar.chaptersButton.hasClass('vjs-hidden'), 'chapters menu is displayed');
+
+  var menuItems = player.controlBar.chaptersButton.items;
+
+  equal(menuItems.length, 2, 'menu contains two item');
+  
+  player.dispose();
+});
+
+test('chapters should be displayed when a track and its cures added and button updated', function() {
+  let player = TestHelpers.makePlayer();
+
+  this.clock.tick(1000);
+  
+  var chapters = player.addTextTrack('chapters', 'Test Chapters', 'en');
+  chapters.addCue({
+    startTime: 0,
+    endTime: 2,
+    text: 'Chapter 1'
+  });
+  chapters.addCue({
+    startTime: 2,
+    endTime: 4,
+    text: 'Chapter 2'
+  });
+  equal(chapters.cues.length, 2);
+
+  player.controlBar.chaptersButton.update();
+  
+  ok(!player.controlBar.chaptersButton.hasClass('vjs-hidden'), 'chapters menu is displayed');
+
+  var menuItems = player.controlBar.chaptersButton.items;
+
+  equal(menuItems.length, 2, 'menu contains two item');
+  
+  player.dispose();
+});
+
+test('chapters menu should use track label as menu title', function() {
+  let player = TestHelpers.makePlayer({
+    tracks: [chaptersTrack]
+  });
+  
+  this.clock.tick(1000);
+  
+  var chapters = player.textTracks()[0];
+  chapters.addCue({
+    startTime: 0,
+    endTime: 2,
+    text: 'Chapter 1'
+  });
+  chapters.addCue({
+    startTime: 2,
+    endTime: 4,
+    text: 'Chapter 2'
+  });
+  equal(chapters.cues.length, 2);
+
+  player.controlBar.chaptersButton.update();
+  
+  var menu = player.controlBar.chaptersButton.menu;
+  var menuTitle = menu.contentEl().firstChild.textContent;
+
+  equal(menuTitle, 'Test Chapters', 'menu gets track label as title');
+  
+  player.dispose();
+});
+
+test('chapters should be displayed when remote track added and load event fired', function() {
+  let player = TestHelpers.makePlayer();
+
+  this.clock.tick(1000);
+  
+  var chaptersEl = player.addRemoteTextTrack(chaptersTrack);
+  
+  chaptersEl.track.addCue({
+    startTime: 0,
+    endTime: 2,
+    text: 'Chapter 1'
+  });
+  chaptersEl.track.addCue({
+    startTime: 2,
+    endTime: 4,
+    text: 'Chapter 2'
+  });
+
+  equal(chaptersEl.track.cues.length, 2);
+
+  TestHelpers.triggerDomEvent(chaptersEl, 'load');
```

Comment:
I think we need another version of this test that uses the emulated text tracks.
```js
let player = TestHelpers.makePlayer({
  techOrder: ['html5'],
  html5: {
    nativeTextTracks: false
  }
});
```
Then we can just call `chaptersEl.trigger('load')` here.